src/shims/posix/fs.rs

```diff
@@ -489,22 +528,32 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             // thus they can share the same implementation here.
             let &[_, _, start] = check_arg_count(args)?;
             let start = this.read_scalar(start)?.to_i32()?;
-            if fd < MIN_NORMAL_FILE_FD {
-                throw_unsup_format!("duplicating file descriptors for stdin, stdout, or stderr is not supported")
-            }
+
             let fh = &mut this.machine.file_handler;
-            let (file_result, writable) = match fh.handles.get(&fd) {
-                Some(file_descriptor) => {
-                    // FIXME: Support "dup" for all FDs(stdin, etc)
-                    let FileHandle { file, writable } = file_descriptor.as_file_handle()?;
-                    (file.try_clone(), *writable)
-                },
-                None => return this.handle_not_found(),
-            };
-            let fd_result = file_result.map(|duplicated| {
-                fh.insert_fd_with_min_fd(FileHandle { file: duplicated, writable }, start)
-            });
-            this.try_unwrap_io_result(fd_result)
+
+            if fd < 3 {
+                let dup_fd : Box<dyn FileDescriptor> = match fd {
+                    0 => Box::new(io::stdin()),
+                    1 => Box::new(io::stdout()),
+                    2 => Box::new(io::stderr()),
```

Comment:
No this is not right. You have to look up the actual FD that is currently at `fd`. It might not be the std one any more!

Reply:
Made changes so that we check if the FileDescriptor is a `File`. If not, then we make a copy of stdin/out/err based on FD.